### DISCUSSION TITLE
Request for Step-by-Step Tutorial or Guide for UniTime

### DISCUSSION THREAD

**Discussion by BingBing160809** [UniTime/unitime#177] [Q&A] [answered]
Hello ,

I have successfully installed UniTime on my server but am having trouble figuring out how to use it. I have been searching for tutorials or step-by-step guides but haven't found anything comprehensive. Could you please provide or direct me to any available resources or documentation that could help me get started with UniTime? Any detailed tutorials or user guides would be greatly appreciated.

Thank you!

**Comment by tomas-muller** [comment]
The UniTime documentation is available at [help.unitime.org/documentation](https://help.unitime.org/documentation). In particular, please take a look at the [administration](https://help.unitime.org/manuals/administration), the [course timetabling data-entry](https://help.unitime.org/manuals/courses-entry), and the [course timetabling solver](https://help.unitime.org/manuals/courses-solver) manuals.

**Reply by BingBing160809** [reply]
Dear  Tomas,

I hope this message finds you well. I am currently working on my Final Year Project (FYP) and would appreciate your guidance regarding the Unitime settings.

Our university operates classes from 8 AM to 6 PM without any break intervals. There are three types of classes: Lectures, Tutorials, and Practical sessions.
Lectures: 3 hours per week, divided into one 120-minute session and one 60-minute session.
Tutorials: One 60-minute session per week.
Practical sessions, Three variations:
 a) One 120-minute session per week,
 b) One 90-minute session per week,
c) One 180-minute session every other week(odd and even).
 I have set up the following configurations.
Could you please confirm if these settings are correct?

I mistakenly set up a "1 x 80" session, but I am unable to locate it to delete it. Could you suggest a quick way to find and remove it?

Despite the above settings, incorrect time display for my class timetable. Could you kindly advise on how to resolve this issue?

I have attached two screenshots for your reference.

Thank you for your time and assistance.
Best regards,
![f36c653aec1766d273168aaa3711af8](https://github.com/user-attachments/assets/9e3f0ef7-9562-4c04-ae1f-c46755f23607)
![faa7fa4e8f6f575423e6c89dcc85065](https://github.com/user-attachments/assets/ce4ace59-5827-4af7-91e4-c496d67a07d3)

Bianca

**Reply by shaftain** [reply]
Time pattern can not be deleted while it is being used. First remove it from all the classes where it is being used and then you can delete it. The other way is to make a new time pattern and use it where needed.

**Reply by tomas-muller** [reply, marked as answer]
The time pattern cannot be deleted as long as it is being used. If it is no longer used in a time preference on a class or a scheduling subpart, it is possible that it was used in a course timetabling solution that is saved. In this case, you can make the time pattern **not visible**, which means it will not appear in the available time patterns dropdown menu.

That said, you can get a list of classes that have an assignment using the time pattern listed using the following HQL (Administration > Utilities > [Test HQL](https://help.unitime.org/test-hql) page):
```
from Assignment a where a.timePattern.name = '1 x 180'
```
Similarly, you can get a list of classes that have the time pattern assigned by using the following query:
```
select c from Class_ c, TimePref p where p.timePattern.name = '1 x 180' and p.owner = c
```
or scheduling subparts
```
select c from SchedulingSubpart c, TimePref p where p.timePattern.name = '1 x 180' and p.owner = c
```

As for your second question, the display on the [Timetable](https://help.unitime.org/timetable) page depends on the selections of the Display Mode, Days, and Times dropdowns. If you select **All** in the Times dropdown, the page will display times from midnight till midnight. You can modify (or create another) time window (Times dropdown in the Filter on the Timetable page) using the `unitime.timeGrid.timesX` application properties (X is a number starting with one). The defaults are:
```
unitime.timeGrid.times1=Daytime|90|210|6
unitime.timeGrid.times2=Evening|210|276|6
unitime.timeGrid.times3=Daytime & Evening|90|276|6
unitime.timeGrid.times4=All|0|288|6
```
The format is name|first slot|last slot|increment, e.g., Daytime|90|222|6 means starting at 7:30 am, ending at 6:30 pm, in half-hour increments. All time slots are 5 minutes long and start at midnight. These properties can be set on the Administration > Defaults > [Configuration](https://help.unitime.org/application-configuration) page.

So, for instance, if you want to change the Daytime & Evening times, create a new application property `unitime.timeGrid.times3` and set it to what you need (using the Administration > Defaults > [Configuration](https://help.unitime.org/application-configuration) page). If you want to create new options, create property `unitime.timeGrid.times5`, `unitime.timeGrid.times6`, and so on.
